submitter/sqs.py

```diff
@@ -1,76 +1,72 @@
+import json
+import logging
+
 import boto3
-import click
+from dspace.client import DSpaceClient
 
+from submitter import config
+from submitter.submission import Submission
 
-def message_loop(input_queue, output_queue, wait):
-    msgs = retrieve(input_queue, wait)
+logger = logging.getLogger(__name__)
 
-    if len(msgs) > 0:
-        click.echo(len(msgs))
-        process(msgs, output_queue)
-        message_loop(input_queue, output_queue, wait)
-    else:
-        click.echo("No messages received")
 
+def message_loop(queue, wait):
+    logger.info("Message loop started")
+    msgs = retrieve(queue, wait)
 
-def process(msgs, output_queue):
-    for message in msgs:
-        click.echo(message.message_attributes)
-        click.echo(message.body)
-        print("Do all the dspace submission stuff here")
+    if len(msgs) > 0:
+        process(msgs)
+        message_loop(queue, wait)
+    else:
+        logger.info("No messages available in queue %s", queue)
 
-        # faking it with always succeeding for now... creating of this status
-        # dict is likely better moved to our upcoming submission class but this
-        # was convenient for initial testing
-        status = {
-            "PackageSource": {
-                "DataType": "String",
-                "StringValue": message.message_attributes.get("PackageSource").get(
-                    "StringValue"
-                ),
-            },
-            "PackageID": {
-                "DataType": "String",
-                "StringValue": message.message_attributes.get("PackageID").get(
-                    "StringValue"
-                ),
-            },
-            "status": {"DataType": "String", "StringValue": "success"},
-            "handle": {
-                "DataType": "String",
-                "StringValue": "https://example.com/handle/this",
-            },
-        }
 
-        # write result to output
-        write(status, output_queue)
+def process(msgs):
+    client = DSpaceClient(config.DSPACE_API_URL)
+    client.login(config.DSPACE_USER, config.DSPACE_PASSWORD)
 
-        # cleanup (probs better to confirm the write to the output was good
-        # before cleanup but for now yolo it)
+    for message in msgs:
+        message_id = message.message_attributes["PackageID"]["StringValue"]
+        message_source = message.message_attributes["SubmissionSource"]["StringValue"]
+        logger.info("Processing message %s from source %s", message_id, message_source)
+        try:
+            submission = Submission.from_message(message)
+        except Exception as e:
+            # TODO: handle and test submit message errors
```

Comment:
This would catch a case where there is something wrong with the format/structure of the message, eg it is missing a field that is required by our spec (as opposed to something being wrong with the contents of the message fields, which would indeed be caught later). The problem with how we handle this (and the reason I didn't just go ahead and implement it) is that on the off chance the field that is missing from the message is the ResultQueue...we don't know where to send the error message. Let's talk about what to do with that. In the meantime I created [a Jira ticket](https://mitlibraries.atlassian.net/browse/ETD-444) for it and I think we can leave it like this for now since we 1) do want to raise the exception and 2) do want to fill in the try/catch block here.